LDAP Single Sign-on

machisuji opened the pull request:
Allows implementing SSO (Single Sign-On) via custom headers passed to OpenProject through a load balancer.

WP [#25671](https://community.openproject.com/projects/gmbh/work_packages/25671/activity)

machisuji committed:
ldap sso

machisuji committed:
localise sso failure error

machisuji committed:
code climate

machisuji committed:
code climate 2

machisuji committed:
code climate 3

machisuji committed:
added missing login param

machisuji committed:
make spec work with SSO module prepended to ApplicationController

oliverguenther reviewed (changes requested):
Overall, looks good to me. There are a few minor things:

1. Setting `auth_source_sso` from configuration.yml yields string keys. This is an error of the configuration parsing, but your module should use `with_indifferent_access` anyway.

2. Accessing `/sso` (e.g., by reloading) results in an error because the session `auth_source_sso_failure` object is not set.

3. Upon errors, two flash messages are being shown.

machisuji committed:
polish

machisuji commented:
@oliverguenther I have addressed your concerns. Calling /sso without a failure will redirect to `/`. The config uses indifferent access. And also there are no longer two flash messages:

<img width="626" alt="screen shot 2017-08-30 at 14 06 48" src="https://user-images.githubusercontent.com/158871/29874050-71c66e1e-8d8d-11e7-840e-d95225c2146c.png">
